xmodule/capa_block.py

```diff
@@ -795,12 +795,21 @@ def generate_report_data(self, user_state_iterator, limit_responses=None):
                 }
                 yield (user_state.username, report)
 
+    def get_course_end_date(self):
+        course_block_key = self.runtime.course_entry.structure['root']
+        return self.runtime.course_entry.structure['blocks'][course_block_key].fields['end']
+
     @property
     def close_date(self):
         """
         Return the date submissions should be closed from.
         """
-        due_date = self.due
+        try:
```

Comment:
Please move this try-except block in `course_end_date` property as suggested above